### PR TITLE
[1616] Use records for all payloads

### LINKED ISSUE #1616: Use records for our payloads

### PR DESCRIPTION
**Pull request by frouene**
Bug: https://github.com/eclipse-sirius/sirius-components/issues/1616

# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [ ] New features
- [ ] Documentation
- [x] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

**Review by gcoutable — CHANGES_REQUESTED** [review]
Is there a need to keep the constructor?
Unless it does something specific, we can remove it as we did for inputs.

**Comment by frouene** [comment]
We keep the constructor to secure the data passed to GraphQL.

File: packages/diagrams/backend/sirius-components-diagrams-graphql/src/main/java/org/eclipse/sirius/components/diagrams/graphql/datafetchers/subscription/DiagramRefreshedEventPayloadDiagramDataFetcher.java [hunk 1]
```diff
@@ -31,8 +31,8 @@ public class DiagramRefreshedEventPayloadDiagramDataFetcher implements IDataFetc
     public DataFetcherResult<Diagram> get(DataFetchingEnvironment environment) throws Exception {
         DiagramRefreshedEventPayload payload = environment.getSource();
         return DataFetcherResult.<Diagram>newResult()
-                .data(payload.getDiagram())
-                .localContext(environment.getLocalContext())
-                .build();
+                                .data(payload.diagram())
+                                .localContext(environment.getLocalContext())
+                                .build();
```
Comment (gcoutable) [inline packages/diagrams/backend/sirius-components-diagrams-graphql/src/main/java/org/eclipse/sirius/components/diagrams/graphql/datafetchers/subscription/DiagramRefreshedEventPayloadDiagramDataFetcher.java]: Did you change the formatting yourself or is it occurred after a save?

File: packages/formdescriptioneditors/backend/sirius-components-formdescriptioneditors-graphql/src/main/java/org/eclipse/sirius/components/formdescriptioneditors/graphql/datafetchers/subscription/FormDescriptionEditorRefreshedEventPayloadFormDescriptionEditorDataFetcher.java [hunk 2]
```diff
@@ -31,8 +31,8 @@ public class FormDescriptionEditorRefreshedEventPayloadFormDescriptionEditorData
     public DataFetcherResult<FormDescriptionEditor> get(DataFetchingEnvironment environment) throws Exception {
         FormDescriptionEditorRefreshedEventPayload payload = environment.getSource();
         return DataFetcherResult.<FormDescriptionEditor>newResult()
-                .data(payload.getFormDescriptionEditor())
-                .localContext(environment.getLocalContext())
-                .build();
+                                .data(payload.formDescriptionEditor())
+                                .localContext(environment.getLocalContext())
+                                .build();
```
Comment (gcoutable) [inline packages/formdescriptioneditors/backend/sirius-components-formdescriptioneditors-graphql/src/main/java/org/eclipse/sirius/components/formdescriptioneditors/graphql/datafetchers/subscription/FormDescriptionEditorRefreshedEventPayloadFormDescriptionEditorDataFetcher.java]: same here

File: packages/forms/backend/sirius-components-forms-graphql/src/main/java/org/eclipse/sirius/components/forms/graphql/datafetchers/subscription/FormRefreshedEventPayloadFormDataFetcher.java [hunk 3]
```diff
@@ -31,8 +31,8 @@ public class FormRefreshedEventPayloadFormDataFetcher implements IDataFetcherWit
     public DataFetcherResult<Form> get(DataFetchingEnvironment environment) throws Exception {
         FormRefreshedEventPayload payload = environment.getSource();
         return DataFetcherResult.<Form>newResult()
-                .data(payload.getForm())
-                .localContext(environment.getLocalContext())
-                .build();
+                                .data(payload.form())
+                                .localContext(environment.getLocalContext())
+                                .build();
```
Comment (gcoutable) [inline packages/forms/backend/sirius-components-forms-graphql/src/main/java/org/eclipse/sirius/components/forms/graphql/datafetchers/subscription/FormRefreshedEventPayloadFormDataFetcher.java]: same here

File: packages/selection/backend/sirius-components-selection-graphql/src/main/java/org/eclipse/sirius/components/selection/graphql/datafetchers/subscription/SelectionRefreshedEventPayloadSelectionDataFetcher.java [hunk 4]
```diff
@@ -31,8 +31,8 @@ public class SelectionRefreshedEventPayloadSelectionDataFetcher implements IData
     public DataFetcherResult<Selection> get(DataFetchingEnvironment environment) throws Exception {
         SelectionRefreshedEventPayload payload = environment.getSource();
         return DataFetcherResult.<Selection>newResult()
-                .data(payload.getSelection())
-                .localContext(environment.getLocalContext())
-                .build();
+                                .data(payload.selection())
+                                .localContext(environment.getLocalContext())
+                                .build();
```
Comment (gcoutable) [inline packages/selection/backend/sirius-components-selection-graphql/src/main/java/org/eclipse/sirius/components/selection/graphql/datafetchers/subscription/SelectionRefreshedEventPayloadSelectionDataFetcher.java]: same here

File: packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/editingcontext/EditingContextQueryBasedIntDataFetcher.java [hunk 5]
```diff
@@ -49,9 +49,9 @@ public CompletableFuture<Integer> get(DataFetchingEnvironment environment) throw
...
 
         return this.editingContextEventProcessorRegistry.dispatchEvent(editingContextId, input)
-                .filter(QueryBasedIntSuccessPayload.class::isInstance)
-                .map(QueryBasedIntSuccessPayload.class::cast)
-                .map(QueryBasedIntSuccessPayload::getResult)
-                .toFuture();
+                                                        .filter(QueryBasedIntSuccessPayload.class::isInstance)
+                                                        .map(QueryBasedIntSuccessPayload.class::cast)
+                                                        .map(QueryBasedIntSuccessPayload::result)
+                                                        .toFuture();
```
Comment (gcoutable) [inline packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/editingcontext/EditingContextQueryBasedIntDataFetcher.java]: same here

File: packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/editingcontext/EditingContextQueryBasedBooleanDataFetcher.java [hunk 6]
```diff
@@ -48,9 +48,9 @@ public CompletableFuture<Boolean> get(DataFetchingEnvironment environment) throw
...
 
         return this.editingContextEventProcessorRegistry.dispatchEvent(editingContextId, input)
-                .filter(QueryBasedBooleanSuccessPayload.class::isInstance)
-                .map(QueryBasedBooleanSuccessPayload.class::cast)
-                .map(QueryBasedBooleanSuccessPayload::getResult)
-                .toFuture();
+                                                        .filter(QueryBasedBooleanSuccessPayload.class::isInstance)
+                                                        .map(QueryBasedBooleanSuccessPayload.class::cast)
+                                                        .map(QueryBasedBooleanSuccessPayload::result)
+                                                        .toFuture();
```
Comment (gcoutable) [inline packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/editingcontext/EditingContextQueryBasedBooleanDataFetcher.java]: same here

File: packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/editingcontext/EditingContextQueryBasedObjectDataFetcher.java [hunk 7]
```diff
@@ -50,11 +50,11 @@ public CompletableFuture<DataFetcherResult<Object>> get(DataFetchingEnvironment
...
-                .filter(QueryBasedObjectSuccessPayload.class::isInstance)
-                .map(QueryBasedObjectSuccessPayload.class::cast)
-                .map(QueryBasedObjectSuccessPayload::getResult)
-                .map(result -> this.toDataFetcherResult(result, editingContextId))
-                .toFuture();
+                                                        .filter(QueryBasedObjectSuccessPayload.class::isInstance)
+                                                        .map(QueryBasedObjectSuccessPayload.class::cast)
+                                                        .map(QueryBasedObjectSuccessPayload::result)
+                                                        .map(result -> this.toDataFetcherResult(result, editingContextId))
+                                                        .toFuture();
```
Comment (gcoutable) [inline packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/editingcontext/EditingContextQueryBasedObjectDataFetcher.java]: same here

File: packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/editingcontext/EditingContextQueryBasedObjectsDataFetcher.java [hunk 8]
```diff
@@ -51,11 +51,11 @@ public CompletableFuture<DataFetcherResult<List<Object>>> get(DataFetchingEnviro
...
-                .filter(QueryBasedObjectsSuccessPayload.class::isInstance)
-                .map(QueryBasedObjectsSuccessPayload.class::cast)
-                .map(QueryBasedObjectsSuccessPayload::getResult)
-                .map(result -> this.toDataFetcherResult(result, editingContextId))
-                .toFuture();
+                                                        .filter(QueryBasedObjectsSuccessPayload.class::isInstance)
+                                                        .map(QueryBasedObjectsSuccessPayload.class::cast)
+                                                        .map(QueryBasedObjectsSuccessPayload::result)
+                                                        .map(result -> this.toDataFetcherResult(result, editingContextId))
+                                                        .toFuture();
```
Comment (gcoutable) [inline packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/editingcontext/EditingContextQueryBasedObjectsDataFetcher.java]: same here

File: packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/editingcontext/EditingContextQueryBasedStringDataFetcher.java [hunk 9]
```diff
@@ -48,9 +48,9 @@ public CompletableFuture<String> get(DataFetchingEnvironment environment) throws
...
 
         return this.editingContextEventProcessorRegistry.dispatchEvent(editingContextId, input)
-                .filter(QueryBasedStringSuccessPayload.class::isInstance)
-                .map(QueryBasedStringSuccessPayload.class::cast)
-                .map(QueryBasedStringSuccessPayload::getResult)
-                .toFuture();
+                                                        .filter(QueryBasedStringSuccessPayload.class::isInstance)
+                                                        .map(QueryBasedStringSuccessPayload.class::cast)
+                                                        .map(QueryBasedStringSuccessPayload::result)
+                                                        .toFuture();
```
Comment (gcoutable) [inline packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/editingcontext/EditingContextQueryBasedStringDataFetcher.java]: same here

File: packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/object/ObjectQueryBasedBooleanDataFetcher.java [hunk 10]
```diff
@@ -54,10 +54,10 @@ public CompletableFuture<Boolean> get(DataFetchingEnvironment environment) throw
...
         QueryBasedBooleanInput input = new QueryBasedBooleanInput(UUID.randomUUID(), query, new HashMap<>(newVariables));
         return this.editingContextEventProcessorRegistry.dispatchEvent(editingContextId, input)
-                .filter(QueryBasedBooleanSuccessPayload.class::isInstance)
-                .map(QueryBasedBooleanSuccessPayload.class::cast)
-                .map(QueryBasedBooleanSuccessPayload::getResult)
-                .toFuture();
+                                                        .filter(QueryBasedBooleanSuccessPayload.class::isInstance)
+                                                        .map(QueryBasedBooleanSuccessPayload.class::cast)
+                                                        .map(QueryBasedBooleanSuccessPayload::result)
+                                                        .toFuture();
```
Comment (gcoutable) [inline packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/object/ObjectQueryBasedBooleanDataFetcher.java]: same here

File: packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/object/ObjectQueryBasedIntDataFetcher.java [hunk 11]
```diff
@@ -55,10 +55,10 @@ public CompletableFuture<Integer> get(DataFetchingEnvironment environment) throw
...
 
         return this.editingContextEventProcessorRegistry.dispatchEvent(editingContextId, input)
-                .filter(QueryBasedIntSuccessPayload.class::isInstance)
-                .map(QueryBasedIntSuccessPayload.class::cast)
-                .map(QueryBasedIntSuccessPayload::getResult)
-                .toFuture();
+                                                        .filter(QueryBasedIntSuccessPayload.class::isInstance)
+                                                        .map(QueryBasedIntSuccessPayload.class::cast)
+                                                        .map(QueryBasedIntSuccessPayload::result)
+                                                        .toFuture();
```
Comment (gcoutable) [inline packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/object/ObjectQueryBasedIntDataFetcher.java]: same here

File: packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/object/ObjectQueryBasedObjectDataFetcher.java [hunk 12]
```diff
@@ -56,11 +56,11 @@ public CompletableFuture<DataFetcherResult<Object>> get(DataFetchingEnvironment
...
-                .filter(QueryBasedObjectSuccessPayload.class::isInstance)
-                .map(QueryBasedObjectSuccessPayload.class::cast)
-                .map(QueryBasedObjectSuccessPayload::getResult)
-                .map(result -> this.toDataFetcherResult(result, editingContextId, newVariables))
-                .toFuture();
+                                                        .filter(QueryBasedObjectSuccessPayload.class::isInstance)
+                                                        .map(QueryBasedObjectSuccessPayload.class::cast)
+                                                        .map(QueryBasedObjectSuccessPayload::result)
+                                                        .map(result -> this.toDataFetcherResult(result, editingContextId, newVariables))
+                                                        .toFuture();
```
Comment (gcoutable) [inline packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/object/ObjectQueryBasedObjectDataFetcher.java]: same here

File: packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/object/ObjectQueryBasedObjectsDataFetcher.java [hunk 13]
```diff
@@ -57,11 +57,11 @@ public CompletableFuture<DataFetcherResult<List<Object>>> get(DataFetchingEnviro
...
-                .filter(QueryBasedObjectsSuccessPayload.class::isInstance)
-                .map(QueryBasedObjectsSuccessPayload.class::cast)
-                .map(QueryBasedObjectsSuccessPayload::getResult)
-                .map(result -> this.toDataFetcherResult(result, editingContextId, newVariables))
-                .toFuture();
+                                                        .filter(QueryBasedObjectsSuccessPayload.class::isInstance)
+                                                        .map(QueryBasedObjectsSuccessPayload.class::cast)
+                                                        .map(QueryBasedObjectsSuccessPayload::result)
+                                                        .map(result -> this.toDataFetcherResult(result, editingContextId, newVariables))
+                                                        .toFuture();
```
Comment (gcoutable) [inline packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/object/ObjectQueryBasedObjectsDataFetcher.java]: same here

File: packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/object/ObjectQueryBasedStringDataFetcher.java [hunk 14]
```diff
@@ -55,10 +55,10 @@ public CompletableFuture<String> get(DataFetchingEnvironment environment) throws
...
 
         return this.editingContextEventProcessorRegistry.dispatchEvent(editingContextId, input)
-                .filter(QueryBasedStringSuccessPayload.class::isInstance)
-                .map(QueryBasedStringSuccessPayload.class::cast)
-                .map(QueryBasedStringSuccessPayload::getResult)
-                .toFuture();
+                                                        .filter(QueryBasedStringSuccessPayload.class::isInstance)
+                                                        .map(QueryBasedStringSuccessPayload.class::cast)
+                                                        .map(QueryBasedStringSuccessPayload::result)
+                                                        .toFuture();
```
Comment (gcoutable) [inline packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/object/ObjectQueryBasedStringDataFetcher.java]: same here

File: packages/trees/backend/sirius-components-trees-graphql/src/main/java/org/eclipse/sirius/components/trees/graphql/datafetchers/subscription/TreeRefreshedEventPayloadTreeDataFetcher.java [hunk 15]
```diff
@@ -31,8 +31,8 @@ public class TreeRefreshedEventPayloadTreeDataFetcher implements IDataFetcherWit
     public DataFetcherResult<Tree> get(DataFetchingEnvironment environment) throws Exception {
         TreeRefreshedEventPayload payload = environment.getSource();
         return DataFetcherResult.<Tree>newResult()
-                .data(payload.getTree())
-                .localContext(environment.getLocalContext())
-                .build();
+                                .data(payload.tree())
+                                .localContext(environment.getLocalContext())
+                                .build();
```
Comment (gcoutable) [inline packages/trees/backend/sirius-components-trees-graphql/src/main/java/org/eclipse/sirius/components/trees/graphql/datafetchers/subscription/TreeRefreshedEventPayloadTreeDataFetcher.java]: same here

**Review by gcoutable — CHANGES_REQUESTED** [review]

File: packages/diagrams/backend/sirius-components-diagrams-graphql/src/main/java/org/eclipse/sirius/components/diagrams/graphql/datafetchers/subscription/DiagramRefreshedEventPayloadDiagramDataFetcher.java (hunk 1 above)
Comment (frouene) [reply packages/diagrams/backend/sirius-components-diagrams-graphql/src/main/java/org/eclipse/sirius/components/diagrams/graphql/datafetchers/subscription/DiagramRefreshedEventPayloadDiagramDataFetcher.java]: It's my formater, it aligns when multiline

Comment (gcoutable) [reply packages/diagrams/backend/sirius-components-diagrams-graphql/src/main/java/org/eclipse/sirius/components/diagrams/graphql/datafetchers/subscription/DiagramRefreshedEventPayloadDiagramDataFetcher.java]: Could you undo all of your formatting changes?

Comment (frouene) [reply packages/diagrams/backend/sirius-components-diagrams-graphql/src/main/java/org/eclipse/sirius/components/diagrams/graphql/datafetchers/subscription/DiagramRefreshedEventPayloadDiagramDataFetcher.java]: Done

**Comment by frouene** [comment]
I changed the format to keep correct alignment.

**Review by gcoutable — APPROVED** [review]

**Pull request merged**